docs/proposals/11-public-prefix.md

```diff
@@ -0,0 +1,155 @@
+- Implement non-disruptive changes for existing IPv4-based setups.
+
+### Non-Goals
+
+- Bring-your-own IPv6 public prefix support is not covered by this proposal. 
+
+## Proposal
+
+Restrict the implementation of the `Prefix` type to only allow `prefixLength` when using the IPv6 IP family. This can 
+be achieved by adding validation logic that prohibits the creation of a non-unique local IPv6 `Prefix` in the `spec.Prefix` 
+field. The allocation of a public routable IPv6 `Prefix` is ensured by a subsequent component and the effective `Prefix`
+is stored in the `Status` of the `Prefix` resource.
+
+The further division of the `Prefix` into sub-allocations via  `spec.parentPrefix` reference allows the user to create 
+`Prefixes` derived from the parent.
+
+For IPv4, no change is needed as we will continue with the existing API contract.
+
+Below is an example of a `Prefix` resource requesting a prefix of size 56:
+
+```yaml
+apiVersion: ipam.ironcore.dev/v1alpha1
+kind: Prefix
+metadata:
+  namespace: default
+  name: my-prefix
+spec:
+  ipFamily: IPv6
+  prefixLength: 56
+status:
+  phase: Pending # Becomes Allocated once the prefix length has been assigned.
+  prefix: ffff::/56 # The effective assigned prefix.
+```
+
+Here is an example of how a subsequent Prefix, which derives its address from a Parent prefix, is defined:
+
+```yaml
+apiVersion: ipam.ironcore.dev/v1alpha1
+kind: Prefix
+metadata:
+  namespace: default
+  name: my-sub-prefix
+spec:
+  ipFamily: IPv6
+  parentPrefix:
+    name: my-prefix
+  prefixLength: 128
+status:
+  phase: Pending # Becomes Allocated once the prefix length has been assigned.
+  prefix: ffff::dddd/128 # The effective assigned prefix.
+```
+
+In the same manner a validation on the `NetworkInterface` resource will ensure that `spec.IPs` only contain unique local
+IPv6 addresses. Alternatively, an ephemeral `Prefix` with a parent relationship to the initial requested `Prefix` ensures 
+a unique public IPv6 address allocation for a `NetworkInterface`.
+
+```yaml
+apiVersion: networking.ironcore.dev/v1alpha1
+kind: NetworkInterface
+metadata:
+  namespace: default
+  name: my-nic
+spec:
+  ips:
+    ephemeral:
+      prefixTemplate:
+        spec:
+          parentRef:
+            name: my-prefix
+```
+
+Here the `prefixTemplate` validation will ensure that no non-unique local IPv6 addresses are assigned.
+
+A similar behaviour is ensured for the internal `LoadBalancer` case. The example below illustrates how an internal
+`LoadBalancer` can request a unique IPv6 address from a parent `Prefix`.
+
+```yaml
+apiVersion: networking.ironcore.dev/v1alpha1
+kind: LoadBalancer
+metadata:
+  namespace: default
+  name: my-lb
+spec:
+  type: Internal
+  ipFamilies:
+    - IPv6
+  ips:
+    - ephemeral:
+        prefixTemplate:
+          spec:
+            parentPrefix:
+              name: my-prefix
+```
+
```

Comment:
What about NAT64 ? 
Do we need to present it on this API level ? @MalteJ 
We already have NAT Gateway object with an ipv4 address, right ? so in case NICs get assigned to that 
NAT Gateway, they will simply use the IPv4 NAT Address for their DNS64 resolved targets ?
No need to mark the NAT Gateway with some ipv6 related info on this API level ?